src/ninshiki_cpp/utils/circle.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2021 ICHIRO ITS
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include "ninshiki_cpp/utils/circle.hpp"
+
+namespace ninshiki_cpp::utils
+{
+
+Circle::Circle(const std::vector<cv::Point> & contour)
+: center(cv::Point2f(-1, -1)), radius(0.0)
+{
+    cv::minEnclosingCircle(cv::Mat(contour), center, radius);
```

Comment:
Can you confirm that this line is working correctly? This documentation said that the first argument of the `cv::minEnclosingCircle` is a list of point instead of an image.

Reply:
`cv::Mat` basically just 2D vector regardless whether it's image or not. In this case, it seems that the input can use `cv::Mat`, not as an image, but as vector of 2d points (vector of 2).